C/eval.c

```diff
@@ -809,56 +812,57 @@ static bool computeEvalTCOBound(memBound *dag_bound, const dag_node* dag, const
     }
   }
 
-  *dag_bound = bound[len-1];
+  {
+    const ubounded inputSize = type_dag[dag[len-1].sourceType].bitSize;
+    const ubounded outputSize = type_dag[dag[len-1].targetType].bitSize;
+    *cellsBound = bounded_add( bounded_add(inputSize, outputSize)
+                             , max(bound[len-1].extraCellsBound[0], bound[len-1].extraCellsBound[1])
+                             );
+    *UWORDBound = (ubounded)ROUND_UWORD(inputSize) + (ubounded)ROUND_UWORD(outputSize)
+                + max(bound[len-1].extraUWORDBound[0], bound[len-1].extraUWORDBound[1]);
+    *frameBound = bound[len-1].extraFrameBound[0] + 2; /* add the initial input and output frames to the count. */
+    *costBound = bound[len-1].cost;
+  }
   free(bound);
-  return true;
+  return (maxCells <= *cellsBound) ? SIMPLICITY_ERR_EXEC_MEMORY
+       : (maxCost <= *costBound) ? SIMPLICITY_ERR_EXEC_BUDGET
```

Comment:
I don't know how I missed it, but these inequalities were strict before and now they are non-strict, which is incorrect. I'll prepare a PR to correct this.

The written specification of "the bound on memory allocation requirements exceed the allowed limits" is just too vague.